satellite/metainfo/metainfo.go

```diff
@@ -631,37 +632,193 @@ func (endpoint *Endpoint) ProjectInfo(ctx context.Context, req *pb.ProjectInfoRe
+		SegmentsSize: req.GetDefaultSegmentSize(),
+		RedundancyScheme: storj.RedundancyScheme{
+			Algorithm:      storj.RedundancyAlgorithm(defaultRS.GetType()),
+			ShareSize:      defaultRS.GetErasureShareSize(),
+			RequiredShares: int16(defaultRS.GetMinReq()),
+			RepairShares:   int16(defaultRS.GetRepairThreshold()),
+			OptimalShares:  int16(defaultRS.GetSuccessThreshold()),
+			TotalShares:    int16(defaultRS.GetTotal()),
+		},
+		EncryptionParameters: storj.EncryptionParameters{
+			CipherSuite: storj.CipherSuite(defaultEP.CipherSuite),
+			BlockSize:   int32(defaultEP.BlockSize),
+		},
+	}, nil
 }
 
 // DeleteBucket deletes a bucket
-func (endpoint *Endpoint) DeleteBucket(ctx context.Context, req *pb.BucketDeleteRequest) (_ *pb.BucketDeleteResponse, err error) {
+func (endpoint *Endpoint) DeleteBucket(ctx context.Context, req *pb.BucketDeleteRequest) (resp *pb.BucketDeleteResponse, err error) {
 	defer mon.Task()(&ctx)(&err)
-	// TODO: placeholder to implement pb.MetainfoServer interface.
-	return &pb.BucketDeleteResponse{}, err
+
+	keyInfo, err := endpoint.validateAuth(ctx, macaroon.Action{
+		Op:            macaroon.ActionDelete,
+		Bucket:        req.Name,
+		EncryptedPath: []byte{},
+		Time:          time.Now(),
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Unauthenticated, err.Error())
+	}
+
+	err = endpoint.metainfo.DeleteBucket(ctx, req.Name, keyInfo.ProjectID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, err.Error())
+	}
+
+	return &pb.BucketDeleteResponse{}, nil
 }
 
-// ListBuckets returns a list of buckets
-func (endpoint *Endpoint) ListBuckets(ctx context.Context, req *pb.BucketListRequest) (_ *pb.BucketListResponse, err error) {
+// ListBuckets returns all buckets in a project
+func (endpoint *Endpoint) ListBuckets(ctx context.Context, req *pb.BucketListRequest) (resp *pb.BucketListResponse, err error) {
 	defer mon.Task()(&ctx)(&err)
-	// TODO: placeholder to implement pb.MetainfoServer interface.
-	return &pb.BucketListResponse{}, err
+	action := macaroon.Action{
+		Op:            macaroon.ActionRead,
+		Bucket:        []byte{},
+		EncryptedPath: []byte{},
+		Time:          time.Now(),
+	}
+	keyInfo, err := endpoint.validateAuth(ctx, action)
+	if err != nil {
+		return nil, status.Errorf(codes.Unauthenticated, err.Error())
+	}
+
+	items, err := endpoint.metainfo.ListBuckets(ctx, keyInfo.ProjectID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "ListBuckets: %v", err)
+	}
+
+	allowedBuckets, err := getAllowedBuckets(ctx, action)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "getAllowedBuckets: %v", err)
+	}
+
+	bucketItems := make([]*pb.BucketListItem, len(items))
+	for i, item := range items {
+		if _, ok := allowedBuckets[item.Name]; ok {
+			bucketItems[i] = &pb.BucketListItem{
+				Name:      []byte(item.Name),
+				CreatedAt: item.Created,
+			}
+		}
+	}
+
+	return &pb.BucketListResponse{
+		Items: bucketItems,
+		More:  false,
```

Comment:
do you plan to support `more` in this PR?

Reply:
yes, in the description of this PR, it says that part is WIP

Reply:
ah, yes :) thanks